### PR TITLE
Enhance Aggregation Fuzzer to test abandon-partial-agg code path

### PR DESCRIPTION
**Pull request by mbasmanova**
The added coverage helped reproduce double-free bug in GroupingSet::toIntermediate.

See #6224

### CODE REVIEW

File: velox/exec/GroupingSet.cpp [hunk 1]
```diff
@@ -1048,17 +1048,19 @@ void GroupingSet::toIntermediate(
       continue;
     }
 
+    // Initialize all groups, even if we only need just one, to make sure bulk
+    // free (intermediateRows_->eraseRows) is safe. It is not legal to free a
+    // group that hasn't been initialized.
+    function->initializeNewGroups(
```
Comment (bikramSingh91) [inline velox/exec/GroupingSet.cpp:1054]: is there any unit test we need to add to cover this fix?

**Review by bikramSingh91 — APPROVED** [review]
Looks good, thanks for adding this.

nit: can you mention explicitly in the commit message that it includes the fix for the bug it found

**Comment by mbasmanova** [comment]
@bikramSingh91 Bikram, thank you for review. I'm landing the fix separately. It is included here because without the fuzzer dies.

File: velox/exec/tests/AggregationFuzzer.cpp [hunk 2]
```diff
@@ -659,21 +697,31 @@ void AggregationFuzzer::go() {
...
+
     if (injectSpill) {
       spillDirectory = exec::test::TempDirectoryPath::create();
       builder.spillDirectory(spillDirectory->path)
           .config(core::QueryConfig::kSpillEnabled, "true")
           .config(core::QueryConfig::kAggregationSpillEnabled, "true")
           .config(core::QueryConfig::kTestingSpillPct, "100");
     }
+
+    if (abandonPartial) {
```
Comment (kagamiori) [inline velox/exec/tests/AggregationFuzzer.cpp:718]: I saw that testAggregations() only tests abandon-partial when grouping key is not empty, but we don't check it here. Does it matter?

Comment (mbasmanova) [reply velox/exec/tests/AggregationFuzzer.cpp:718]: @kagamiori Good catch, Wei. I'll make a change to not test this path for global aggregations. The code path we are targeting doesn't kick in for global aggregations, hence, testing is just wasted valuable time and compute. Thanks.

**Pull request closed without merging**